app/code/Magento/CatalogGraphQl/Model/Resolver/Products.php

```diff
@@ -118,14 +134,28 @@ public function resolve(
             );
         }
 
+        $options = $this->catalogConfig->getAttributeUsedForSortByArray();
```

Comment:
Please create a separate resolver for `sort_fields`. There are 2 main reasons for that:
1. Performance. In case `sort_fields` is not requested, all this logic will not be executed in case of separate resolver
2. Following single responsibility and object decomposition simplifies maintenance and understanding

Reply:
Roger, i will fix it, thanks for your feedback